### PR TITLE
feat: Add filtering by collection

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it allows to filter by collection. 

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: packages/infrastructure/src/private/algolia/search/infrastructure/get-facets-infra.ts [hunk 1]
```diff
@@ -30,6 +30,18 @@ export const algoliaGetFacetsInfra = ({
           acc.push(`categories.lvl0:'${formattedValue}'`);
         }
 
+        if (name === "collection") {
+          const formattedValue = value
+            .split(".")
+            .map(
+              (v) =>
+                `'${v.charAt(0).toUpperCase() + v.slice(1).replaceAll("-", " & ")}'`,
```
Comment (wojtek007mirumee) [inline packages/infrastructure/src/private/algolia/search/infrastructure/get-facets-infra.ts:38]: you add space also. Please confirm it is intended change.

**Review by wojtek007mirumee — APPROVED** [review]
